### ISSUE TITLE
[Week 5] Preparing for v1.0 & Graded LO 5.11

### ISSUE THREAD

**Issue by okkhoy** [nusCS2113-AY1819S1/forum#18]
# Project v1.0 submission

- Ensure that you submit the document (**a single PDF file with all parts in it**) on IVLE on time. 
- The submission should have the following three parts.
  - Requirements (In the form of user stories, use cases and non-functional requirements).
  - User guide (Use a format similar to [AddressBook-Level4 user guide](https://github.com/nusCS2113-AY1819S1/addressbook-level4/blob/master/docs/UserGuide.adoc)).
  - A draft of the plan for your project (This can be updated later!).
    - Each team member should identify the feature (s)he is developing.
    - Each feature should be divided into three increments, to be released at v1.1, v1.2 and v1.3.
    - Plan in such a way that each increment is end-user testable.
- Name the file as `{Team ID}.pdf` (e.g., `W12-1.pdf`)

# Graded LO 5.11 - Enhancing Addressbook-Level2

### Individual
- Submit a PR to the semester fork of [Addressbook-Level2](https://github.com/nusCS2113-AY1819S1/addressbook-level2).
- Your PR should include these things
  - Your enhancement (i.e., code changes)
  - Updated user guide and (optionally) developer guide
  - Updated tests

### Team
- Review PRs created by (at least one of) your team members

# Git config

- Ensure that, on your computers you set the `git config user.name` to the `GitHub username` you provided in the pre-module survey.
- We will depend on this config information to associate the work done by you while awarding participation points for the coding exercises and to track your project progress.

**Comment by QzSG** [comment]
Are we allowed to modify or remove requirements after this milestone or is it cast in stone? I am referring to things like adding user stories/ features. Also, can members developing a certain feature change? (Before the next milestone that is)

**Comment by okkhoy** [comment]
- Yes! Requirements can be modified later. 
  - Unless you change the direction of the project, you can retain the user stories and just reduce their priority.
  - You can always add new user stories, but don't do this until the end of the semester!! 😜 
- Yes! Members can also change the feature they are developing. 
  - Again don't do this in the last minute. Give yourself time to actually develop the feture.

**Comment by QzSG** [comment]
@okkhoy Do you mind elaborating on
> Unless you change the direction of the project

**Comment by 0WN463** [comment]
If the User guide is meant to be for v2.0 (ie the theoretical end product), there will include features that we do not implement in v1.4. Do we also need to write how to use these "would-be unimplemented" functions in the User Guide? If so, the User Guide would be quite bloated with unimplemented features

**Comment by okkhoy** [comment]
> Unless you change the direction of the project ...

We are pretty flexible about what you want to implement, so it may not be so uncommon that you started off with something, but after a week or so, you decide to change the features so much that it may end up being a different product!! So in this case, the project direction is considered changed

E.g., You start off to morph Addressbook to task manager, but you can potentially end up doing a time-table creation/manager app!

**Comment by okkhoy** [comment]
> Do we also need to write how to use these "would-be unimplemented" functions in the User Guide?

Yes! but be brief about it! This gives us an idea that you have thought the product through thoroughly!
